### DISCUSSION TITLE
Debugger 2.0 (Dec 3rd)

### DISCUSSION THREAD

**Discussion by sang-yeo** [twilio-labs/paste#987] [Office hours] [answered]
Background: we want customers to easily locate new errors and access the information they need
Stage of Design: Hi-fi
Link to prototype: [link](https://twilio.invisionapp.com/share/NUZLGG8W9BG#/439248474_R5-_cover) (PW:1122)

I'd like to get feedback on overall components usage and overall interaction

**Comment by SiTaggart** [comment]
Hi @sang-yeo 

This is a pretty big flow of many screens. I don't think a 20 minute slot is an appropriate amount of time to review an entire feature in.

So to help us out, do you have specific questions regarding component usage you'd like to cover? Any that caused concerns in testing? Any that cause concern with the dev team? Any specific areas that you are unsure if you have used the right component for the right interaction? Anything that didn't have enough documentation on?

**Reply by sang-yeo** [reply]
Yes, I agree this is a big flow of many screen. Lets me list it out. 

- I heard that you will release the table pagination on Dec. I'd like to apply it to my design. Could you tell me where to get it? 
- Do you recommend Export(It might change it to Download) button with icon? 
- Do I use destructive button for 'Clear filter' and 'Reset'? 
- I'd like to get feedback on the Next and Previous buttons I used in the "Request inspector" modal

**Reply by SiTaggart** [reply]
So much better! Thank you! 🙏

**Comment by serifluous** [comment, marked as answer]
> I heard that you will release the table pagination on Dec. I'd like to apply it to my design. Could you tell me where to get it?

It's still in development, but it will likely look pretty close to this (for table pages that have different URLs):

<img width="825" alt="Screen Shot 2020-12-03 at 11 40 11 AM" src="https://user-images.githubusercontent.com/3821840/101079663-9ea7ea00-355c-11eb-9a22-fad9c67189a5.png">

Here's the in-progress [design](https://share.goabstract.com/476d75d6-59de-4278-a46f-4708e08fe174?mode=design&sha=9e3aa0e28f0a83ed155776d12f08f4b7ed51f127) and [spec](https://docs.google.com/document/d/1ciGnSQmkWLnkKr6h3KEEPvbehchX_sv-VmJ5QHZIs88/edit?usp=sharing) if you want to see how it's shaping up.

> Do you recommend Export(It might change it to Download) button with icon?

This may have to be a link instead of a button (see Discussion #848), so I'd recommend reaching out to your engineers. For the icon: If you find through user testing that it helps add clarity, add the icon! If there's no meaningful difference, you don't need it.

See the last paragraph in our ["Composing a Button" guidelines](https://paste.twilio.design/components/button/#composing-a-button) (it applies to anchors too): "_Pair text with an icon only if the icon clarifies the meaning of the button._"

> Do I use destructive button for 'Clear filter' and 'Reset'?

I assume it's pretty easy for a customer to redo the filters if they accidentally clear them? And it has no effect on anything outside of filtering the table on the page? [Destructive buttons](https://paste.twilio.design/components/button/#destructive-secondary-button) should only be used when the action "could be difficult to reverse"

> I'd like to get feedback on the Next and Previous buttons I used in the "Request inspector" modal

imo the up and down arrows could almost seem like you're expanding or shrinking the modal (but that could just be me). You could use the Pagination component here (even with the left/right arrows) in a Modal Footer. We've never seen this kind of Modal interaction/layout before, so using Pagination here would also be worth testing. If any customer issues show up because of the component, we'd love to know, too!

**Reply by sang-yeo** [reply]
@serifluous Thanks for answering all my questions. 
<img width="1008" alt="Screen Shot 2020-12-03 at 2 15 00 PM" src="https://user-images.githubusercontent.com/67388401/101095587-00734e80-3573-11eb-8637-9b6eb57e6d8d.png">

In this case, a user only has one event, seems like showing the total number of the event under the table is visually off. What do you suggest?

**Reply by serifluous** [reply]
You could probably remove/hide the pagination component entirely in a case like this since you have the "Showing 1 of 1" text at the top of the table.

But I'll take this case into consideration as we continue work on Pagination design. I have a hunch we'll have the component just push that "1 of 1" text all the way to the right edge of the table—not completely sure yet though.

**Comment by richbachman** [comment]
Hi @sang-yeo 

Other than @serifluous's comments, the only other thing I'd add pertains to: 

> Do you recommend Export(It might change it to Download) button with icon?

I think the word Export makes the most sense here since the user would be getting the logs in a different format. It looks like @ctecks gives similar feedback here: https://github.com/twilio-labs/paste/discussions/990#discussioncomment-145789, which you may have already heard about.
